### ISSUE TITLE
Code implementation does not seem to match Conditional Coverage Loss from paper

### ISSUE THREAD

**Issue by chrisyeh96** [Achenred/End-to-end-CRO#1]
In the [arXiv paper](https://arxiv.org/abs/2403.04670), the gradient for the conditional coverage loss $\hat L_{CC}$ includes the term

$$
    \nabla_\phi g_{\phi^*(\mathcal{U}_\theta)}(\psi^i)
$$

which suggests that the loss involves calculating gradients w.r.t. the weights $\phi$ of the model $g$. My understanding is that $g$ corresponds to the scikit-learn Support Vector Classifier instantiated via these lines in the code:

https://github.com/Achenred/End-to-end-CRO/blob/0f544f02d96a3da410c8844d33ce6475dade1f66/utils.py#L476-L477

Given that this is a scikit-learn model as opposed to a PyTorch model, how do you pass PyTorch gradients through $\nabla_\phi g$ after you've already trained $g$?

More generally, could you please explain how the equation for $\nabla_\theta \hat{L}_{CC}$ from your paper matches what you have implemented in your code?
